### PR TITLE
Add interfaces related to function details

### PR DESCRIPTION
**Pull request by ignacy123**

### CODE REVIEW

File: app/src/main/kotlin/compiler/intermediate_form/FunctionDetailsGenerator.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import compiler.ast.Variable
+
+data class FunctionDetailsGenerator(
+    val depth: Int,
+    val vars: Map<Variable, Boolean>,
+    val parameters: List<Variable>
+)
+
+fun generateCall(args: List<IntermediateFormTreeNode>): ControlFlowGraph {
```
Comment (ggawryal) [inline app/src/main/kotlin/compiler/intermediate_form/FunctionDetailsGenerator.kt:11]: Shouldn't we also return an IFTNode for function result, outside that CFG: `Pair<ControlFlowGraph, IntermediateFormTreeNode?>`?

File: app/src/main/kotlin/compiler/intermediate_form/FunctionDetailsGenerator.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+fun genPrologue(): ControlFlowGraph {
+    return TODO()
+}
+
+fun genEpilogue(): ControlFlowGraph {
+    return TODO()
+}
+
+fun genRead(variable: Variable, direct: Boolean): IntermediateFormTreeNode {
```
Comment (JHodorHodor) [inline app/src/main/kotlin/compiler/intermediate_form/FunctionDetailsGenerator.kt:23]: `isDirect`?

**Review by JHodorHodor — APPROVED** [review]
LGTM

File: app/src/main/kotlin/compiler/intermediate_form/FunctionDetailsGenerator.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import compiler.ast.Variable
+
+data class FunctionDetailsGenerator(
+    val depth: Int,
+    val vars: Map<Variable, Boolean>,
+    val parameters: List<Variable>
+)
+
+fun generateCall(args: List<IntermediateFormTreeNode>): Pair<ControlFlowGraph, IntermediateFormTreeNode?> {
```
Comment (JHodorHodor) [inline app/src/main/kotlin/compiler/intermediate_form/FunctionDetailsGenerator.kt:11]: Maybe instead of a pair, we should use an object with named fields?

**Pull request merged**